src/Database/Mysql/Driver/Mysql.php

```diff
@@ -136,13 +136,15 @@ protected function retrieveTables(Tenant $tenant): array
         $tempTenant = $tenant->replicate();
         $tempTenant->{$tenant->getTenantKeyName()} = $tenant->getOriginal($tenant->getTenantKeyName());
 
+        $tempConnectionName = Str::random(16);
+
         /** @var ResolvesConnections $resolver */
         $resolver = resolve(ResolvesConnections::class);
-        $resolver($tempTenant, Tenancy::getTenantConnectionName());
+        $resolver($tempTenant, $tempConnectionName);
 
-        $tables = Tenancy::getTenantConnection()->getDoctrineSchemaManager()->listTableNames();
+        $tables = DB::connection($tempConnectionName)->getDoctrineSchemaManager()->listTableNames();
 
-        $resolver(null, Tenancy::getTenantConnectionName());
+        $resolver(null, $tempConnectionName);
```

Comment:
Can we please simplify this variable into `$connection`? It is obvious with the `Str::random` that this concerns a random connection name.